### PR TITLE
Added generate_helix and integrated into Edge Class

### LINKED ISSUE #5: Populate an Edge with a DNA single/double Helix
## Summary

Within the code we have something that represents an edge, most probably a class called `Edge`.

We now need to use this edge to act as a template for a strand of DNA that can wrap around it.

There is a repository found here: https://github.com/lorenzo-rovigatti/tacoxDNA - this contains conversions to/from the oxDNA file format.

Also this file: https://github.com/rgatkinson/oxdna/blob/master/UTILS/generators.py shows how to generate an oxDNA DNA strand with the geometry of the single/double helix

We basically need a function within/subclass of `Edge` that contains the important information needed to generate the oxDNA file + further, this must be created for each nucleotide (3 columns each, 15 columns in total)

- [ ] position (**centre-of-mass**/base/backbone [not sure which])
- [ ] a_1 vector
- [ ] a_3 vector
- [ ] linear velocity (can be initialised to 0.0 to start)
- [ ] angular velocity (can be initialised to 0.0 to start)

Further, we need to create a topology file that describes the bonding. For this we need the following information (1 column each)

- [ ] strand number
- [ ] base (e.g. 'A', 'T', 'C', 'G')
- [ ] nucleotide before (3' direction)
- [ ] nucleotide after (5' direction)

### PR DESCRIPTION
**Pull request by shanilpanara**
Still need to clean up generate_helix code and add docstrings.

But now an edge creates its own `system` when initialised.

Can now close #5 

@debeshmandal I'm sure there will be clashes with your updates to the code, if its ok with you, I want to work through these tomorrow so I get some experience resolving conflicts.

### CODE REVIEW

**Comment by debeshmandal** [comment]
I like it - some points::

- Each `Edge` should have a function (not property since there are many parameters that we can change e.g. sequence) called `strand()` - which means the following:
  - the `Edge` shouldn't create a `System`
  - for the purposes of keeping the implementation I think that the `nucleotides` property of `Edge` should be removed - an `Edge` should be purely geometrical (not related to DNA) but with a sole 'gateway' to a DNA system by using the `strand()` function - this is a more broad point that is subjective and up for discussion

File: origamiUROP/oxdna/strand.py [hunk 1]
```diff
@@ -42,6 +42,10 @@ def nucleotides(self) -> list:
             nucleotide._strand_index = self.index
         return self._nucleotides
 
+    def add_nucleotide(self, Nucleotide: Nucleotide):
```
Comment (debeshmandal) [inline origamiUROP/oxdna/strand.py:45]: don't use `Nucleotide` since this is already declared as a class - use `nucleotide`

**Comment by debeshmandal** [comment]
The `generate_helix` function should be in the `strand.py` file (to help tidy things up)

**Comment by shanilpanara** [comment]
#### Main things from today:
- sorted out passing a sequence when calling `generateHelix`, this took a lot longer than it needed to!
- added testing alongside this for both the `generateHelix()` function and `Edge.strand()` method
> Each Edge should have a function (not property since...

- Also worked on this, now we can pass any other parameters (which are passed into the generateHelix function) to the method in the `Edge` class called `strand()`
- We can then access this `Strand` object stored within the `Edge` object as `EdgeObject.helix`

@debeshmandal

**Comment by debeshmandal** [comment]
@shanilpanara sounds good - have you pushed through everything yet? when you have, feel free to merge and then I will make some edits if necessary. After a quick glance, it looks great and any edits I make will be mostly cosmetic 👍

**Comment by shanilpanara** [comment]
Everything is pushed that I have, will merge now @debeshmandal

**Pull request merged**